server/neptune/workflows/internal/terraform/job/runner.go

```diff
@@ -176,3 +180,13 @@ func (r *jobRunner) runOptionalSteps(ctx *job.ExecutionContext, localRoot *root.
 
 	return nil
 }
+
+// Executes the TerraformCloseJob activity
+func (r *jobRunner) closeTerraformJob(ctx *job.ExecutionContext) {
+	var resp string
+
+	// let's not fail this workfklow if closing the job fails since it's not critical to close the job for the tf workflow
+	_ = workflow.ExecuteActivity(ctx, r.Activity.TerraformCloseJob, activities.TerraformCloseJobRequest{
+		JobID: ctx.JobID,
+	}).Get(ctx, &resp)
```

Comment:
Isn't this technically critical though? I mean we can't just completely drop our logs and have no way of recovering them right?  Additionally, its not like you can apply a plan without seeing the logs, that could be bad.

Either way right now this is going to retry indefinitely and block so it doesn't matter that you're not returning an error, it would get stuck here.

Reply:
Ahh I could have worded it better! I meant it's not critical to persist the job yet since we could persist it on shutdown if it fails for some reason. 

I have added logs to the activity when closing job fails. Even if we fail to close the job, the job will still be in memory since we only remove it if we have successfully persisted it. Upon shutdown, the temporal worker will try to persist all the jobs in memory to ensure no logs are lost. I can also add stats around job failures to alarm us if we see a spike.